CONTRIBUTING.md

```diff
@@ -1,12 +1,30 @@
 # Contributing
 
+Firstly, thanks for taking the time to contribute! Here's hopefully some helpful
+info to get you started.
+
+## Before Adding a Feature
+
+Before a new feature or endpoint is added to this repo you'll want to make sure
+that there is agreement amongst the main parties using BSP. In order for this to
+happen please make sure there is an
+[Issue](https://github.com/build-server-protocol/build-server-protocol/issues)
+or a
+[Discussion](https://github.com/build-server-protocol/build-server-protocol/discussions/landing)
```

Comment:
I would turn this sentence into a recommendation instead of a requirement.

For some small changes, I feel like it can be a waste of time to do it in two steps: open an issue, wait for approval, then work on the PR, wait for a second approval. A PR with a good description can save time of both the contributor and the reviewer, even if there is a chance the PR get rejected.

Reply:
Ah, so I more meant when you add an entire new feature like `buildTarget/formatting` or something that you do need a discussion. I figured small improvements, etc implicitly don't need one. I'll make this clearer in there.